services/pgnotifier/pgnotifier.go

```diff
@@ -108,10 +108,10 @@ func loadPGNotifierConfig() {
 	pgNotifierDBPort = config.GetInt("PGNOTIFIER_DB_PORT", 5432)
 	pgNotifierDBPassword = config.GetString("PGNOTIFIER_DB_PASSWORD", "ubuntu") // Reading secrets from
 	pgNotifierDBSSLMode = config.GetString("PGNOTIFIER_DB_SSL_MODE", "disable")
-	config.RegisterIntConfigVariable(3, &maxAttempt, false, 1, "PgNotifier.maxAttempt")
+	config.RegisterIntVar(3, &maxAttempt, 1, "PgNotifier.maxAttempt")
 	trackBatchInterval = time.Duration(config.GetInt("PgNotifier.trackBatchIntervalInS", 2)) * time.Second
-	config.RegisterDurationConfigVariable(5000, &maxPollSleep, true, time.Millisecond, "PgNotifier.maxPollSleep")
-	config.RegisterDurationConfigVariable(120, &jobOrphanTimeout, true, time.Second, "PgNotifier.jobOrphanTimeout")
+	config.RegisterAtomicDurationVar(5000, &maxPollSleep, time.Millisecond, "PgNotifier.maxPollSleep")
+	config.RegisterAtomicDurationVar(120, &jobOrphanTimeout, time.Second, "PgNotifier.jobOrphanTimeout")
```

Comment:
Good catch.

Reply:
Did we get this while running the test with `-race`?

Reply:
This has already been take care of with the new changes. These globals have been [removed](https://github.com/rudderlabs/rudder-server/pull/3784/files#diff-81c88462ea98fe7900948848a9b8e201fd33f6efdf9c6770de3cbd64e657bab9R64-R65).

Reply:
Yes the `-race` highlighted it :+1: I'll try to have a look at that PR too then so maybe we can speed this up. With the atomic change I'm consistently getting failing tests now.

Reply:
> This has already been take care of with the new changes. These globals have been [removed](https://github.com/rudderlabs/rudder-server/pull/3784/files#diff-81c88462ea98fe7900948848a9b8e201fd33f6efdf9c6770de3cbd64e657bab9R64-R65).

@achettyiitr the race can still happen in your branch as well. if the variable is hot-reloaded while we read then you get a race. it's just going to be harder to see this race in tests because the variable is not global anymore.

Reply:
Understood. Hot reloadable then need to concurrent safe then